lib/dor/services/client/marcxml.rb

```diff
@@ -21,6 +23,39 @@ def catkey(barcode:)
 
           raise_exception_based_on_response!(resp)
         end
+
+        # Gets MARCXML corresponding to a barcode or catkey
+        # @param barcode [String] required string representing a barcode
+        # @param catkey [String] required string representing a catkey
+        # @raise [NotFoundResponse] when the response is a 500 with "Record not found in Symphony"
+        # @raise [UnexpectedResponse] on an unsuccessful response from the server
+        # @return [String] MARCXML
+        def marcxml(barcode: nil, catkey: nil)
+          check_args(barcode, catkey)
+
+          resp = connection.get do |req|
+            req.url "#{api_version}/catalog/marcxml"
+            req.params['barcode'] = barcode unless barcode.nil?
+            req.params['catkey'] = catkey unless catkey.nil?
+          end
+
+          raise_exception_based_on_marcxml_response!(resp)
```

Comment:
I might suggest putting `return resp.body if resp.success?` *before* the `raise_exception...` line. That means you no longer have to check `resp.success` twice in the raise method.

Reply:
In my opinion, having the method end with the happy path (return the body) rather than the unhappy path (raising) makes it easier to understand the method.

Reply:
OK, no problem.